### PR TITLE
HHH-18513 fix wrong prePartialFlush calculation in Session Metrics log

### PR DESCRIPTION
**Pull request by boskoop**
[HHH-18513](https://hibernate.atlassian.net/browse/HHH-18513): Session Metrics for pre-partial-flushes are wrong

`StatisticalLoggingSessionEventListener` was incorrectly calculating the pre-partial-flush duration (probably due to a variable typo that passed review in commit e4e26ee989aaab34b27e5ced491d8cf4a91cf1e2).

----------------------
By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt)
and can be relicensed under the terms of the [LGPL v2.1 license](https://www.gnu.org/licenses/old-licenses/lgpl-2.1.txt) in the future at the maintainers' discretion.
For more information on licensing, please check [here](https://github.com/hibernate/hibernate-orm/blob/main/CONTRIBUTING.md#legal).

----------------------

[HHH-18513]: https://hibernate.atlassian.net/browse/HHH-18513?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by boskoop** [comment]
It seems that there was an error in the database setup for the oracle test. The tests fail with a "cannot connect to database" error (which also seems unrelated to the proposed change):
```
SessionFactoryTesting > initializationError FAILED
    org.hibernate.service.spi.ServiceException: Unable to create requested service [org.hibernate.engine.jdbc.env.spi.JdbcEnvironment] due to: Error calling Driver.connect() [ORA-12514: Cannot connect to database. Service freepdb1 is not registered with the listener at host localhost port 1521.
```

There was also an error during the "Start database" task:
```
SQL> ALTER DATABASE OPEN
*
ERROR at line 1:
ORA-03113: end-of-file on communication channel
Process ID: 386
Session ID: 162 Serial number: 21621
Help: https://docs.oracle.com/error-help/db/ora-03113/


Elapsed: 00:00:00.86
SQL> SQL> SQL> ERROR:
ORA-03114: not connected to ORACLE
Help: https://docs.oracle.com/error-help/db/ora-03114/
```

Is this a temporary/flaky error in the oracle database setup? If so, could you please retry the build job? Apparently, I don't have the rights to do so.

File: hibernate-core/src/test/java/org/hibernate/engine/internal/StatisticalLoggingSessionEventListenerTest.java [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+		String sessionMetricsLog = messages.get( 0 );
+
+		// acquiring JDBC connections
+		SessionMetric acquiringJdbcConnectionsMetric = extractMetric(
+				sessionMetricsLog,
+				"([0-9]+) nanoseconds spent acquiring ([0-9]+) JDBC connections"
+		);
+		assertThat( acquiringJdbcConnectionsMetric.getDuration() ).isGreaterThan( 0L )
+				.isLessThan( sessionNanoDuration );
+		assertThat( acquiringJdbcConnectionsMetric.getCount() ).isEqualTo( 1 );
```
Comment (boskoop) [inline hibernate-core/src/test/java/org/hibernate/engine/internal/StatisticalLoggingSessionEventListenerTest.java:87]: Is it ok to check for the actual count metrics or do you expect them to be too unstable? Certain changes in the internal logic could result in test failures here.

File: hibernate-core/src/test/java/org/hibernate/engine/internal/StatisticalLoggingSessionEventListenerTest.java [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+				performingL2CPutsMetric,
+				performingL2CHitsMetric,
+				performingL2CMissesMetric,
+				executingFlushesMetric,
+				executingPrePartialFlushesMetric,
+				executingPartialFlushesMetric
+		);
+		assertThat( metricList.size() ).isEqualTo( numberOfMetrics );
+		// Number of lines
+		assertThat( sessionMetricsLog.lines().count() ).isEqualTo( numberOfMetrics + 2 );
```
Comment (boskoop) [inline hibernate-core/src/test/java/org/hibernate/engine/internal/StatisticalLoggingSessionEventListenerTest.java:196]: This is done in order to make the test fail if new metrics are added to log event.

Comment (dreab8) [reply hibernate-core/src/test/java/org/hibernate/engine/internal/StatisticalLoggingSessionEventListenerTest.java:196]: I would remove this check or add a message that makes easy to understand what is causing the failure

File: hibernate-core/src/test/java/org/hibernate/engine/internal/StatisticalLoggingSessionEventListenerTest.java (hunk 1 above)
Comment (dreab8) [reply hibernate-core/src/test/java/org/hibernate/engine/internal/StatisticalLoggingSessionEventListenerTest.java:87]: probably better to avoid this check

**Comment by dreab8** [comment]
Thanks @boskoop ,

apart the couple of comments on the test the PR looks good to me.

File: hibernate-core/src/test/java/org/hibernate/engine/internal/StatisticalLoggingSessionEventListenerTest.java (hunk 1 above)
Comment (boskoop) [reply hibernate-core/src/test/java/org/hibernate/engine/internal/StatisticalLoggingSessionEventListenerTest.java]: I removed the count checks

File: hibernate-core/src/test/java/org/hibernate/engine/internal/StatisticalLoggingSessionEventListenerTest.java (hunk 2 above)
Comment (boskoop) [reply hibernate-core/src/test/java/org/hibernate/engine/internal/StatisticalLoggingSessionEventListenerTest.java]: I improved the failure message. It should be clear now that we check for the number of lines in the log to be equal to the number of metrics plus header and footer line

**Comment by dreab8** [comment]
Thanks @boskoop

**Comment by dreab8** [comment]
squashed in on commit and applied manually.

**Pull request closed without merging**